sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -363,6 +403,8 @@ RT::PiProgram ProgramManager::loadProgram(OSModuleHandle M,
     F.close();
   }
   // Load the selected image
+  if (!is_compiler_available(Context, Format))
+    throw feature_not_supported("Online compilation is not supported in this context");
```

Comment:
`loadProgram` do not compile anything, so it's not clear why `loadProgram` throws in this case.

Please, leave only one check:
1. Either OpenCL API version/extension. 
2. Online compilation support.

Current implementation of `is_compiler_available` checks both and this seems to be unnecessary as the second check is "dead code".

IMO, the most clean way to implement is to go with (1) here and move (2) to place, where we use online compilation (which I believe is already done by https://github.com/intel/llvm/pull/317).

If you are going to have the check for the compiler in this method, please, add a comment clarifying SPIR-V format requires JIT compilation and we check if online compiler is available before we create a program from SPIR-V format. And remove the second patch adding OpenCL API version checks from this PR.

Reply:
loadProgram will indeed compile from SPIR-V to 'native code' when a SPIR-V image is in the file.  I originally had only the is_compiler_available code, but I was asked to add the check for OpenCL version as well.  I will change it to only check the OpenCL version

Reply:
I think I have it right now.  Please check again